### ISSUE TITLE
Add support for annualized data.

### ISSUE THREAD

**Issue by hobuobi** [Recidiviz/justice-counts-components#80]
Corresponding backend ticket is https://github.com/Recidiviz/recidiviz-data/issues/6644

**PRD:** [Bookmark](https://docs.google.com/document/d/1eOiMzk8_eDjC4W0xn89qlkmj_6SOqi1dBTyEusAGGaI/edit?ts=5fb45edc#bookmark=id.wgq2v7cyggo7)
**Mocks:** [Figma](https://www.figma.com/file/66HyjH0efZMoAuTlhnojWh/Justice-Counts-Report?node-id=1130%3A19)

Currently, we only show data that is either "point-in-time" (like populations), or aggregated monthly. However, this excludes a lot of the **yearly-aggregated data** that we actually store in the database, giving the user a sense that there is less data available publicly than there actually is. At the same time, we want to prioritize showing monthly data where we can. 

Our solution for this will be to implement an "annual" switch that allows users to see all metrics aggregated to an annual statistic. This switch should exist at the top of the page; switching it should result in the following changes. 

![Group 70 (1)](https://user-images.githubusercontent.com/7134733/112909087-97759080-90a5-11eb-845e-a05296f33cb5.png)

## Metric Blocks 
- A percent change should still show up on these metric blocks. 
- The date range shown on the block should now correspond to the start and end dates for the aggregated data. For example, if the data for Parole Revocations was aggregated between July 2018 and June 2019, that is the date range that should appear on the block.
- For any population counts (Prison, Probation, Parole), use the population from the last month in the date range of interest (whatever is used in the flow diagram). 

## Flow Diagram 
- The primary date range should correspond with the aggregation range for metrics that are annualized in the original report where possible. 
    - If there are multiple metrics that are originally reported on an annual time range but they have conflicting time ranges, the primary date range should align with the majority.
- For any population counts (Prison, Probation, Parole), use the population from the last month in the date range of interest. 
- In some cases, monthly data will be artificially restricted in order to better match data that is only reported annually. In this case, we should indicate this on the card by including a warning tooltip with the following copy and design:
> Data is restricted to older reports in order to match the other reports shown; however, more recent data is available (as of January 2021). Use the Reporting Frequency switch at the top of the page to see the most recent monthly data. 

![Group 71 (3)](https://user-images.githubusercontent.com/7134733/113660950-029d0500-965a-11eb-9a7b-e8e57ef0f6f0.png)

As a bonus, the orange link in the design above ("Reporting Frequency switch") should scroll the user back to the page position where the data reporting frequency link can be found.

## Graphs
- If possible, bar graphs should be used instead of line graphs for annualized data. 
- Metrics should still appear on the X-axis aligned with a month/year. The month/year should be the last month/year within the aggregation range (e.g., if a particular data point was generated from event counts between January 2019 and December 2019, the date for that metric should be 12/2019).
- The “1 year” option should **not** be present for annual data. Thus, the only time range options for annual data should be “5 years” and “All Time”.

![Group 65 (1)](https://user-images.githubusercontent.com/7134733/112908961-52515e80-90a5-11eb-822d-747a8db75d91.png)

**Comment by colincadams** [comment]
@hobuobi two questions:

1. Should the header above the flow chart be the end month of the primary date range?
2. Should the population graphs stay monthly or move to annual bars?

I think I've asked these before but can't remember the answers 😬 

@nasaownsky I think that it will make sense to have a separate data file for the annual view! I'll try to add details in a bit, but the format should be very similar to the existing file.

**Comment by hobuobi** [comment]
> Should the header above the flow chart be the end month of the primary date range?
Yes!

> Should the population graphs stay monthly or move to annual bars?
Bars represent the annualized value, and show the end-month of the annual aggregation range!

**Comment by colincadams** [comment]
Okay @hobuobi one more question 😁  We did discuss this in [this thread](https://docs.google.com/document/d/1eOiMzk8_eDjC4W0xn89qlkmj_6SOqi1dBTyEusAGGaI/edit?disco=AAAALuT8Nq4), however there we decided to simply omit data that uses a different year. I forgot this when creating the fixture file, and CO reports probation populations at a different cadence, so we ended up with the following situation:

![Screen Shot 2021-04-06 at 11 02 25 AM](https://user-images.githubusercontent.com/3762913/113757847-f51e6400-96c7-11eb-91c8-c95171fe4f01.png)

Here the primary date is September 2020, but we don't have probation population for September so we use June 2020 instead. Based on our discussion in the thread we should omit this data, and in this case we would end up with a tooltip that tells us there is data on the monthly tab even though we omit it here. However, there are cases where the data would be omitted from both pages (annual data for a different period) which does not seem ideal.

Some possible options:
* Omit the data (as discussed in thread in PRD)
* Current behavior from screenshot (misalignment shown in flowchart, but lump it in with September data in graph)
* Current behavior + change labels to only show years on the graphs
* Current behavior + add tooltip to graph legend as well ([this frame from mocks](https://www.figma.com/file/66HyjH0efZMoAuTlhnojWh/Justice-Counts-Report?node-id=1001%3A863))

Sorry for re-hashing this, happy to proceed with just omitting it, but thought it might be useful to look at it with real data
